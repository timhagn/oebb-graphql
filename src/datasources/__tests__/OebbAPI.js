const OebbAPI = require(`../OebbAPI`)

const oebbAPI = new OebbAPI()

describe('[OebbAPI]', () => {
  it('isHealthy() returns true', async () => {
    const healthy = await oebbAPI.isHealthy()
    expect(healthy).toBeTruthy()
  })

  it('getLocations() returns empty array without name', async () => {
    const locations = await oebbAPI.getLocations()
    expect(locations).toMatchInlineSnapshot(`Array []`)
  })

  it('getLocations() returns empty array with empty name', async () => {
    const name = ``
    const locations = await oebbAPI.getLocations(name)
    expect(locations).toMatchInlineSnapshot(`Array []`)
  })

  // it('getLocations() returns empty array with wrong name', async () => {
  //   const name = "@"
  //   const locations = await oebbAPI.getLocations(name)
  //   expect(locations).toMatchInlineSnapshot(`Array []`)
  // })

  it('getLocations() returns array with name "Wien"', async () => {
    const name = `Wien`
    const locations = await oebbAPI.getLocations(name)
    expect(locations).toMatchSnapshot()
  })
})
