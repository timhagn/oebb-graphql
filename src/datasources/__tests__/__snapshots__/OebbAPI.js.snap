// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[OebbAPI] getLocations() returns array with name "Wien" 1`] = `
Array [
  Object {
    "geo": Object {
      "id": "1190100",
      "latitude": 48.208547,
      "longitude": 16.372134,
      "type": "location",
    },
    "id": "1190100",
    "name": "Wien",
    "products": Object {
      "bus": true,
      "ferry": false,
      "interregional": true,
      "national": true,
      "nationalExpress": true,
      "onCall": true,
      "regional": true,
      "suburban": true,
      "subway": false,
      "tram": true,
    },
    "type": "stop",
  },
  Object {
    "geo": Object {
      "id": "1290401",
      "latitude": 48.185507,
      "longitude": 16.375326,
      "type": "location",
    },
    "id": "1290401",
    "name": "Wien Hbf (U)",
    "products": Object {
      "bus": true,
      "ferry": false,
      "interregional": true,
      "national": true,
      "nationalExpress": true,
      "onCall": false,
      "regional": true,
      "suburban": true,
      "subway": true,
      "tram": true,
    },
    "type": "stop",
  },
  Object {
    "geo": Object {
      "id": "1191201",
      "latitude": 48.174559,
      "longitude": 16.333211,
      "type": "location",
    },
    "id": "1191201",
    "name": "Wien Meidling",
    "products": Object {
      "bus": true,
      "ferry": false,
      "interregional": true,
      "national": true,
      "nationalExpress": true,
      "onCall": false,
      "regional": true,
      "suburban": true,
      "subway": true,
      "tram": true,
    },
    "type": "stop",
  },
  Object {
    "geo": Object {
      "id": "1291201",
      "latitude": 48.174433,
      "longitude": 16.331926,
      "type": "location",
    },
    "id": "1291201",
    "name": "Wien Meidling Bahnhof (U)",
    "products": Object {
      "bus": true,
      "ferry": false,
      "interregional": true,
      "national": true,
      "nationalExpress": true,
      "onCall": false,
      "regional": true,
      "suburban": true,
      "subway": true,
      "tram": true,
    },
    "type": "stop",
  },
  Object {
    "geo": Object {
      "id": "1490191",
      "latitude": 48.208709,
      "longitude": 16.372305,
      "type": "location",
    },
    "id": "1490191",
    "name": "Wien Zentrum",
    "products": Object {
      "bus": true,
      "ferry": false,
      "interregional": false,
      "national": false,
      "nationalExpress": false,
      "onCall": false,
      "regional": true,
      "suburban": true,
      "subway": true,
      "tram": false,
    },
    "type": "stop",
  },
]
`;
